extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/StaticResourcesProcessor.java

```diff
@@ -35,15 +36,19 @@ public class StaticResourcesProcessor {
     @BuildStep
     void collectStaticResources(Capabilities capabilities,
             List<AdditionalStaticResourceBuildItem> additionalStaticResources,
-            BuildProducer<StaticResourcesBuildItem> staticResources) {
+            BuildProducer<StaticResourcesBuildItem> staticResources,
+            LaunchModeBuildItem launchModeBuildItem) {
         if (capabilities.isPresent(Capability.SERVLET)) {
             // Servlet container handles static resources
             return;
         }
         Set<StaticResourcesBuildItem.Entry> paths = getClasspathResources();
-        for (AdditionalStaticResourceBuildItem bi : additionalStaticResources) {
-            paths.add(new StaticResourcesBuildItem.Entry(bi.getPath(), bi.isDirectory()));
+        if (!launchModeBuildItem.getLaunchMode().isDevOrTest()) {
```

Comment:
Probably add a comment explaining where they're added in the case of dev/test mode?